### PR TITLE
working litmus restarts_on_kill test

### PR DESCRIPTION
**Pull request by williscool**
cncf/cnf-conformance#158

shout out to denver's inital work for cncf/cnf-conformance#157

- Use correct LIB_IMAGE for chaos test
- Rollback halite version sure to bugs
- Add inital code for chaos network loss

# CNF Conformance PR Template

## Description
Container kill chaos test

## Issues:
Refs: #158 

## How has this been tested:
 - [ ] Covered by existing integration testing
 - [ ] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

### CODE REVIEW

**Comment by williscool** [comment]
TODOS

- finish rest of ac stuff i.e. test writing

File: shard.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ targets:
   cnf_conformance:
     main: src/cnf_conformance.cr
 
-crystal: 0.31.1
+crystal: 0.34.0
```
Comment (taylor) [inline shard.yml:13]: Please switch to 0.33 re: #242

File: src/tasks/resilience.cr [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      resp = upsert_passed_task("restarts_on_kill","✔️  PASSED: container-kill chaos test passed 🗡️💀♻️")
+    else
+      resp = upsert_failed_task("restarts_on_kill","✖️  FAILURE: container-kill chaos test failed 🗡️💀♻️")
+    end
+
+    resp
+  end
+end
+
+def chaos_template__restarts_on_kill
```
Comment (taylor) [inline src/tasks/resilience.cr:83]: please update name from `chaos_template__restarts_on_kill` to `chaos_template__chaos_container_kill`

File: src/tasks/resilience.cr [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    until verdict
+      verdict = `kubectl get chaosresults.litmuschaos.io conformance-container-kill -o jsonpath='{.status.experimentstatus.verdict}'` 
+      puts verdict if check_verbose(args)  
+    end
+
+    puts `#{describe_chaos_result}` if check_verbose(args)  
+
+    if verdict == "Pass"
+      resp = upsert_passed_task("restarts_on_kill","✔️  PASSED: container-kill chaos test passed 🗡️💀♻️")
```
Comment (taylor) [inline src/tasks/resilience.cr:74]: rename `restarts_on_kill` to `chaos_container_kill`

File: src/tasks/resilience.cr [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    else
+      resp = upsert_failed_task("restarts_on_kill","✖️  FAILURE: container-kill chaos test failed 🗡️💀♻️")
```
Comment (taylor) [inline src/tasks/resilience.cr:76]: rename `restarts_on_kill` to `chaos_container_kill`

File: src/tasks/resilience.cr [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    if errors < 1
+      template = Crinja.render(chaos_template__restarts_on_kill, { 
+        "test_name" => "conformance-container-kill" , "helm_chart_container_name" => "#{helm_chart_container_name}", "deployment_label" => "#{deployment_label}" , "deployment_label_value" => "#{deployment_label_value}" 
+      })
+      chaos_config = `echo "#{template}" > "#{destination_cnf_dir}/container-kill-chaosengine.yml"`
+      puts "#{chaos_config}" if check_verbose(args)
+      run_chaos = `kubectl create -f "#{destination_cnf_dir}/container-kill-chaosengine.yml"`
+      puts "#{run_chaos}" if check_verbose(args)
+    else
+      resp = upsert_failed_task("restarts_on_kill","✖️  FAILURE: No deployment label found for container kill test")
```
Comment (taylor) [inline src/tasks/resilience.cr:57]: rename `restarts_on_kill` to `chaos_container_kill`

File: src/tasks/resilience.cr [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+    errors = 0
+    begin
+      deployment_label_value = deployment.get("metadata").as_h["labels"].as_h[deployment_label].as_s
+    rescue ex
+      errors = errors + 1
+      puts ex.message 
+    end
+    if errors < 1
+      template = Crinja.render(chaos_template__restarts_on_kill, {
```
Comment (taylor) [inline src/tasks/resilience.cr:49]: rename `restarts_on_kill` to `chaos_container_kill`

File: src/tasks/resilience.cr [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+task "install_litmus" do |_, args|
+    # TODO: only install if not already installed
+    litmus_install = `kubectl create -f https://litmuschaos.github.io/pages/litmus-operator-latest.yaml`
+    puts "#{litmus_install}" if check_verbose(args)
+    #ait for deployment
+end
+
+desc "Does the CNF come back up when the container is killed"
+# task "restarts_on_kill", ["install_litmus", "retrieve_manifest"] do |_, args|
+task "restarts_on_kill", ["retrieve_manifest"] do |_, args|
```
Comment (taylor) [inline src/tasks/resilience.cr:23]: rename `restarts_on_kill` to `chaos_container_kill`

File: src/tasks/resilience.cr [hunk 8]
```diff
@@ -0,0 +1,116 @@
+# coding: utf-8
+require "sam"
+require "colorize"
+require "crinja"
+# require "./utils/utils.cr"
+
+desc "The CNF conformance suite checks to see if the CNFs are resilient to failures."
+task "resilience", ["restarts_on_kill"] do |t, args|
```
Comment (taylor) [inline src/tasks/resilience.cr:8]: rename `restarts_on_kill` to `chaos_container_kill`

File: points.yml [hunk 9]
```diff
@@ -98,3 +98,6 @@
 
 - name: k8s_conformance
   tags: platform, dynamic
+
+- name: restarts_on_kill
```
Comment (taylor) [inline points.yml:102]: rename `restarts_on_kill` to `chaos_container_kill`

**Comment by williscool** [comment]
think im going to drop ltimus for this if I can get chaos mesh working.

think I figured out the issue at least on my local machine

under the hood litmus uses https://github.com/alexei-led/pumba

and for whatever reason it doesn't pick up the correct `DOCKER_HOST` env var

```

2020-06-04T05:20:02Z" level=error msg="failed to list containers" app=pumba error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" function=github.com/sirupsen/logrus.WithError source="container/client.go:90"
time="2020-06-04T05:20:02Z" level=error msg="failed to list containers" app=pumba error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" function=github.com/sirupsen/logrus.WithError source="docker/kill.go:86"
time="2020-06-04T05:20:02Z" level=error msg="failed to run chaos command" app=pumba error="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" function=github.com/sirupsen/logrus.WithError source="chaos/command.go:79"

```

**Comment by williscool** [comment]
deprecated in favor of

https://github.com/cncf/cnf-conformance/pull/262

**Comment by ksatchit** [comment]
Sorry for commenting on the closed thread! Support for this has been added with recent releases in litmus. We will be setting up a local env/contributing a working version of these tests in the coming weeks - in case you'd be interested in looking at it.

**Pull request closed without merging**